pkg/sources/filesystem/filesystem_test.go

```diff
@@ -120,3 +113,192 @@ func TestScanFile(t *testing.T) {
 
 	assert.Contains(t, foundSecret, secretPart1+secretPart2)
 }
+
+func TestEnumerate(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+
+	// Setup the connection to test enumeration.
+	units := []string{
+		"/one", "/two", "/three",
+		"/path/to/dir/", "/path/to/another/dir/",
+	}
+	conn, err := anypb.New(&sourcespb.Filesystem{
+		Paths:       units[0:3],
+		Directories: units[3:],
+	})
+	assert.NoError(t, err)
+
+	// Initialize the source.
+	s := Source{}
+	err = s.Init(ctx, "test enumerate", 0, 0, true, conn, 1)
+	assert.NoError(t, err)
+
+	reporter := sourcestest.TestReporter{}
+	err = s.Enumerate(ctx, &reporter)
+	assert.NoError(t, err)
+
+	assert.Equal(t, len(units), len(reporter.Units))
+	assert.Equal(t, 0, len(reporter.UnitErrs))
+	for _, unit := range reporter.Units {
+		assert.Contains(t, units, unit.SourceUnitID())
+	}
```

Comment:
Do you think it's worth checking the opposite of this as well? (I'm presuming that there's no handy collection comparison function available.)

Reply:
Yeah, since one is a `[]string` and the other is a `[]SourceUnit`, it was easier to check this way. The longer more correct way would be to convert the `[]SourceUnit` to `[]string` and check that they match.. I was being lazy.